chall07/vsempere.go

```diff
@@ -0,0 +1,59 @@
+package main
+
+import (
+	"fmt"
+	"strings"
+	"strconv"
+	"regexp"
+	"log"
+	"io/ioutil"
+	"net/http"
+)
+
+func toHex(dec []byte) string {
+	ri, err := strconv.Atoi(string(dec))
+	if err != nil {
+		log.Fatal(err)
+	}
+	str := strconv.FormatInt(int64(ri), 16)
+	if len(str) < 2 {
+		return "0"+ str
+	} else {
+			return str
+	}
+}
+
+func launchGet(url string) (string, string) {
+	res, err := http.Get(url)
+	if err != nil {
+		log.Fatal(err)
+	}
+	robots, err := ioutil.ReadAll(res.Body)
+	res.Body.Close()
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	return res.Status, string(robots);
+}
+
+func buildResponseUrl(response string) string {
+	re := regexp.MustCompile(`id=(\d+),r=(\d+),g=(\d+),b=(\d+).*here:\s+(.*)`)
+	m := re.FindSubmatch([]byte(response));
+
+	id := m[1]
+	r := toHex(m[2])
+	g := toHex(m[3])
+	b := toHex(m[4])
+	s := []string{r, g, b}
```

Comment:
A most simple way of doing this would have been by using [fmt.Sscanf](https://golang.org/pkg/fmt/#Sscanf). As you know exactly how the input should be:
```suggestion
	var id, r, g, b int
	_, err := fmt.Sscanf(response, "id=%d,r=%d,g=%d,b=%d - Send your response here: chall07.42madrid.com/?id=<id>&resp=<hex>", &id, &r, &g, &b); err != nil {
		log.Fatal(err)
	}